util/qubes/qubes.lisp

```diff
@@ -0,0 +1,77 @@
+;;;; qubes.lisp
+
+(in-package #:qubes)
+
+;; Qubes module for StumpWM.
+;;
+;; Copyright (C) 2016 Johanna Abrahamsson
+;;
+;; Maintainer: Johanna Abrahamsson
+;;
+
+;;; Code:
+
+(defun alloc-color-on-current (color)
+  (stumpwm::alloc-color (current-screen) color))
+
+
+(setf *label-colors* (list (alloc-color-on-current "pink1")
+						   (alloc-color-on-current "red1") 
+						   (alloc-color-on-current "orange1") 
+						   (alloc-color-on-current "yellow1") 
+						   (alloc-color-on-current "green1") 
+						   (alloc-color-on-current "gray1") 
+						   (alloc-color-on-current "blue1") 
+						   (alloc-color-on-current "purple1") 
+						   (alloc-color-on-current "black")))
+
+
+(defun window-label-color (window)
+  (nth (or (first (xlib:get-property (window-xwin window) :_QUBES_LABEL)) 0) *label-colors*))
+
+
+(defun window-vm (window)
+  (let ((wmproperty (xlib:get-property (window-xwin window) :_QUBES_VMNAME)))
+  (if (= (length wmproperty) 0) nil (stumpwm::utf8-to-string wmproperty))))
+
+(defun window-wmname (window)
+  (let ((wmproperty (xlib:get-property window :WM_NAME)))
+  (if (= (length wmproperty) 0) nil (stumpwm::utf8-to-string wmproperty))))
+
+
+(defmethod stumpwm::update-decoration ((stumpwm::window stumpwm::tile-window))
+  ;; give it a colored border but only if there are more than 1 frames.
+  (let* ((group (window-group window))
+         (screen (group-screen group)))
+				 (let ((c (or (window-label-color window) (screen-unfocus-color screen))))
+      (setf (xlib:window-border (window-parent window)) c
+            ;; windows that dont fill the entire screen have a transparent background.
+            (xlib:window-background (window-parent window))
+            (if (eq (window-type window) :normal)
+                (if (eq *window-border-style* :thick)
+                    c
+                    (screen-win-bg-color screen))
+                :none))
+      ;; get the background updated
+      (xlib:clear-area (window-parent window)))))
+
+(defun stumpwm::xwin-name (win)
+(stumpwm::escape-caret (or
```

Comment:
This line needs to be indented two spaces, and the forms inside the `or` should be aligned. I'd go with this:

```lisp
(defun stumpwm::xwin-name (win)
  (stumpwm::escape-caret (or (stumpwm::xwin-net-wm-name win)
                             (xlib:wm-name win)
                             (window-wmname win))))
```